### PR TITLE
Route-editing + waypoint view fields

### PR DESCRIPTION
**Pull request by ginold**
small update for route-editing form and wp view.

answers #277  and #280

### CODE REVIEW

**Comment by asaunier** [comment]
Please use more explicit commit messages like "fix attributes incorrectly shown in route editing form" or "Move access_period to the route text attributes block".
Else we won't have much info about the changes when looking at the commits history.

I have just tested the route editing form: the extra MTB fields no longer show up when selecting skitouring. But neither when selecting MTB?

Thanks for the change about access_period!

**Comment by asaunier** [comment]
Please merge if 
>  the extra MTB fields no longer show up when selecting skitouring. But neither when selecting MTB?

is addressed.

**Comment by ginold** [comment]
i've made a few other small changes -> i used the mako syntax in for loops instead of angular, 
changed url to a <a href> link, deleted a field that i created but didn't even exist, added other one missing.. maybe you want to have a quick look.

But i'm still waiting to clarify the 'orientation(s)' question. I also still see some strange errors while saving some attributes, but i don't know if I should correct it or is it an API problem?

For example : 
`['multi', 'bloc'] is not a string: {'climbing_outdoor_types': ''}` or `"None" is not iterable` even if it can be null and is not required.

**Comment by asaunier** [comment]
>  i'm still waiting to clarify the 'orientation(s)' question

See https://github.com/c2corg/v6_api/pull/227 (and in common)

> "None" is not iterable even if it can be null and is not required.

This seems to be a problem with the API, see https://github.com/c2corg/v6_ui/issues/110#issuecomment-208739547

> ['multi', 'bloc'] is not a string: {'climbing_outdoor_types': ''}

Are you sure that the error?
I guess the API expects to receive:
```
'climbing_outdoor_types': ['multi', 'bloc']
```

File: c2corg_ui/templates/route/edit.html [hunk 1]
```diff
@@ -80,7 +80,7 @@ <h2 class="heading show-phone" translate>general informations</h2>
                 <p ng-message="required" class="label label-danger" style="{{ (editForm.lang.$invalid && editForm.lang.$touched) ? 'display:inline' : '' }}" translate>This field is required.</p>
               </div>
             </div>
-            
+            {{route.orientations}}
```
Comment (asaunier) [inline c2corg_ui/templates/route/edit.html]: Is this really needed?

File: c2corg_ui/templates/route/helpers/detailed_route_attributes.html [hunk 2]
```diff
@@ -2,22 +2,20 @@
...
-  % if route.get('areas') or route.get('orientation'):
+  % if route.get('areas') or route.get('orientations'):
     <div class="name-icon-value location" ng-click="detailsCtrl.openTab($event)">
       <h4><span translate>Location</span><span class="glyphicon glyphicon-menu-right"></span></h4>
       <div class="name-icon">
         <span class="glyphicon glyphicon-map-marker"></span>
         <p translate>Location</p>
       </div>
-      <span class="detail-text accordion">
+      <span class="detail-text accordion" id="detail-orientation" ng-init="route.orientation = ${route.get('orientations')}">
```
Comment (asaunier) [inline c2corg_ui/templates/route/helpers/detailed_route_attributes.html]: Missing "s" in ``ng-init="route.orientation``?

File: c2corg_ui/templates/route/helpers/detailed_route_attributes.html [hunk 3]
```diff
@@ -223,21 +240,23 @@
 
 ## ACCESS
 <%def name="get_route_access(route)">\
-  % if route.get('lift_access') or route.get('height_diff_access'):
+  % if route.get('lift_access') == False or route.get('lift_access') == True or route.get('height_diff_access'):
```
Comment (asaunier) [inline c2corg_ui/templates/route/helpers/detailed_route_attributes.html]:
```
% if route.get('lift_access') not None or route.get('height_diff_access'):
```

File: c2corg_ui/templates/route/helpers/detailed_route_attributes.html [hunk 4]
```diff
@@ -223,21 +240,23 @@
...
       <div class="name-icon">
         <span class="glyphicon glyphicon-road"></span>
         <p translate>Access</p>
       </div>
       <span class="detail-text accordion">
-        % if route.get('lift_access') and route['lift_access'] is not 'None':
-          <b translate class="value-title">lift_access</b>
-          <input type="checkbox" checked="checked"><span></span>
+
+        % if route.get('lift_access') == True:
```
Comment (asaunier) [inline c2corg_ui/templates/route/helpers/detailed_route_attributes.html]:
```
% if route.get('lift_access'):
```

File: c2corg_ui/templates/route/helpers/detailed_route_attributes.html [hunk 5]
```diff
@@ -223,21 +240,23 @@
...
+        <span x-translate class="value-title">lift_access</span> : <span translate class="value">yes</span><br>
+        % elif route.get('lift_access') == False :
```
Comment (asaunier) [inline c2corg_ui/templates/route/helpers/detailed_route_attributes.html]:
```
% elif route.get('lift_access') not None:
```

File: c2corg_ui/templates/waypoint/helpers/detailed_waypoint_attributes.html [hunk 6]
```diff
@@ -300,29 +316,34 @@
...
+        
       </span>
     </div>
   % endif
 </%def>
 
 ## EQUIPMENT  
 <%def name="get_waypoint_equipment(waypoint)">\
-  % if waypoint.get('matress_unstaffed') or  waypoint.get('gas_unstaffed') or waypoint.get('heating_unstaffed') or waypoint.get('blanket_unstaffed') :
+  % if waypoint.get('matress_unstaffed') == True or waypoint.get('matress_unstaffed') == False or  \
```
Comment (asaunier) [inline c2corg_ui/templates/waypoint/helpers/detailed_waypoint_attributes.html]:
```
% if waypoint.get('matress_unstaffed') not None or  \
```
ditto for the following conditions.

**Comment by ginold** [comment]
> I guess the API expects to receive:

`'climbing_outdoor_types': ['multi', 'bloc']` <

that's exactly what i'm sending. I understand it as if API was awaiting a string rather than an array, or maybe JSON.stringified array but that makes little sense.

**Comment by asaunier** [comment]
Indeed, I have the same behaviour.

> I understand it as if API was awaiting a string rather than an array

Actually the API indeed expects a string:
https://github.com/c2corg/v6_api/blob/master/c2corg_api/models/route.py#L148
not a list of strings. Ditto in the google sheet,

But that consistent: a route cannot be of several types at the same time!

**Comment by tsauerwein** [comment]
If `climbing_outdoor_types` is not an array it should be named `climbing_outdoor_type`.

**Comment by ginold** [comment]
that's exactly what is coming up next in one of Alex'es pr!

File: requirements.txt [hunk 7]
```diff
@@ -14,6 +14,6 @@ Markdown==2.6.5
 # c2corg_common project
 # for development use a local checkout
 # -e ../v6_common
-git+https://github.com/c2corg/v6_common.git@343e1ef
+git+https://github.com/c2corg/v6_common.git@b8e402d
```
Comment (asaunier) [inline requirements.txt]: The hash for c2corg_common with the fixes you suggested about some route/WP attributes is f825f8f

**Comment by asaunier** [comment]
Except for the hash of c2corg_common, your changes look good to me too.

**Comment by ginold** [comment]
I changed the hash. May i merge after travis has passed?

**Comment by asaunier** [comment]
:+1:

**Pull request merged**